splunk_topology/test_splunk_topology.py

```diff
@@ -1237,3 +1478,607 @@ def _mocked_extract_components(*args, **kwargs):
+        })
+
+        self.assertEqual(instances[0]['components'][1], {
+            "externalId": u"server_2",
+            "type": {"name": u"server"},
+            "data": {
+                u"description": u"My important server 2",
+                u"_time": u"2017-03-06T14:55:54.000+00:00",
+                "label.label2Key": "label2Value",
+                "tags": ['result_tag2', 'mytag', 'mytag2']
+            }
+        })
+
+        self.assertEquals(instances[0]["start_snapshot"], True)
+        self.assertEquals(instances[0]["stop_snapshot"], True)
+
+        self.assertEquals(self.service_checks[0]['status'], 0, "service check should have status AgentCheck.OK")
+        # clear the in memory token
+        self.check.status.data.clear()
+        self.check.status.persist("splunk_topology")
+
+    def test_check_memory_token_expired(self):
+        """
+            Splunk topology check should fail when memory token is expired itself.
+        """
+
+        config = {
+            'init_config': {},
+            'instances': [
+                {
+                    'url': 'http://localhost:8089',
+                    'authentication': {
+                        'basic_auth': {
+                            'username': "admin",
+                            'password': "admin"
+                        },
+                        'token_auth': {
+                            'name': "api-admin",
+                            'initial_token': "dsfdgfhgjhkjuyr567uhfe345ythu7y6tre456sdx",
+                            'audience': "admin",
+                            'renewal_days': 10
+                        }
+                    },
+                    'component_saved_searches': [{
+                        "name": "components",
+                        "parameters": {}
+                    }],
+                    'relation_saved_searches': [],
+                    'tags': ['mytag', 'mytag2']
+                }
+            ]
+        }
+
+        self.load_check(config)
+        self.check.status.data['http://localhost:8089token'] = "dsvljbfovjsdvkj"
+        self.check.status.persist("splunk_topology")
+
+        def _mocked_is_token_expired(*args):
+            return True
+
+        self.run_check(config, mocks={
+            '_dispatch_saved_search': _mocked_dispatch_saved_search,
+            '_search': _mocked_search,
+            '_saved_searches': _mocked_saved_searches,
+            'is_token_expired': _mocked_is_token_expired,
+        })
+
+        msg = "Current in use authentication token is expired. Please provide a valid token in the YAML and restart" \
+              " the Agent"
+        # Invalid token should throw a service check with proper message
+        self.assertEquals(self.service_checks[0]['status'], 2, msg)
+
+        # clear the in memory token
+        self.check.status.data.clear()
+        self.check.status.persist("splunk_topology")
+
+    def test_check_initial_token_flag_false_after_creation(self):
```

Comment:
I notice that this test is not in metric and not in event, do we need that there as well?

Reply:
I can add this test even in metric/event as well.